src/main/java/org/molgenis/downloader/api/metadata/Tag.java

```diff
@@ -61,29 +72,30 @@ public void setLabel(String label) {
 
   @Override
   public boolean equals(Object o) {
-    if (this == o) return true;
-    if (!(o instanceof Tag)) return false;
-
-    Tag tag = (Tag) o;
-
-    if (label != null ? !label.equals(tag.label) : tag.label != null) return false;
-    if (objectIRI != null ? !objectIRI.equals(tag.objectIRI) : tag.objectIRI != null) return false;
-    if (relationIRI != null ? !relationIRI.equals(tag.relationIRI) : tag.relationIRI != null)
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
       return false;
-    if (relationLabel != null
-        ? !relationLabel.equals(tag.relationLabel)
-        : tag.relationLabel != null) return false;
-    return codeSystem != null ? codeSystem.equals(tag.codeSystem) : tag.codeSystem == null;
+    }
+    Tag tag = (Tag) o;
+    return Objects.equals(getLabel(), tag.getLabel())
+        && Objects.equals(getObjectIRI(), tag.getObjectIRI())
+        && Objects.equals(getRelationIRI(), tag.getRelationIRI())
+        && Objects.equals(getRelationLabel(), tag.getRelationLabel())
+        && Objects.equals(getValue(), tag.getValue())
+        && Objects.equals(getCodeSystem(), tag.getCodeSystem());
   }
 
   @Override
   public int hashCode() {
-    int result = label != null ? label.hashCode() : 0;
-    result = 31 * result + (objectIRI != null ? objectIRI.hashCode() : 0);
-    result = 31 * result + (relationIRI != null ? relationIRI.hashCode() : 0);
-    result = 31 * result + (relationLabel != null ? relationLabel.hashCode() : 0);
-    result = 31 * result + (codeSystem != null ? codeSystem.hashCode() : 0);
-    return result;
+    return Objects.hash(
+        getLabel(),
```

Comment:
no more need of null checks?

Reply:
I think the `Objects.hash()` takes care of that

Reply:
did you test?

Reply:
Well... no. I trust IntelliJ 🙏

Reply:
It's handled by `Arrays.hashCode()`:

```
    public static int hashCode(Object a[]) {
        if (a == null)
            return 0;

        int result = 1;

        for (Object element : a)
            result = 31 * result + (element == null ? 0 : element.hashCode());

        return result;
    }